libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx

```diff
@@ -70,12 +82,16 @@ export function NetworkingPortSettingModal({ port, onClose, onSubmit }: Networki
             control={control}
             rules={{
               required: 'Please enter service name.',
+              ...namePatternRules,
             }}
             render={({ field, fieldState: { error } }) => (
               <InputText
                 name={field.name}
                 onChange={(e: FormEvent<HTMLInputElement>) => {
-                  setValue('name', `${e.currentTarget.value}-p${watchInternalPort}`)
+                  const name = `${e.currentTarget.value}-p${watchInternalPort}`
+                  if (name.length < 38) {
```

Comment:
Can't you use `maxlength` props on the input directly?

Reply:
I don't find anything in the library and documentation, but I can do something like this: https://github.com/Qovery/console/pull/1444/commits/69debcbf4d31480d0a25dd42d0881eddca36d5d4

Reply:
I was thinking about the html props https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength
no library involve

Reply:
I tried to add `maxLength` on the input but the `setValue` don't take into account this rule. It works only if I edit the input directly not from an `setValue`